app/controllers/projects_controller.rb

```diff
@@ -318,8 +322,8 @@ def fork
       redirect_to child.urlbase
       # todo - notifications
     else
-      flash[:alert] = "Couldn't fork project, try again."
-      redirect_to @project
+      flash[:alert] = "Couldn't fork project, you already have a project with the same name."
```

Comment:
@EmilyDirsh at that point, you'd see an error page not the normal page with a flash message :) Anyways, I'm planning to do it in a better way :)